contracts/adapters/Managing.sol

```diff
@@ -96,23 +117,22 @@ contract ManagingContract is IManaging, DaoConstants, MemberGuard {
                 data,
                 msg.sender
             );
-        _sponsorProposal(dao, proposalId, data, sponsoredBy, votingContract);
-    }
-
-    function _sponsorProposal(
-        DaoRegistry dao,
-        bytes32 proposalId,
-        bytes memory data,
-        address sponsoredBy,
-        IVoting votingContract
-    ) internal {
         dao.sponsorProposal(proposalId, sponsoredBy);
         votingContract.startNewVotingForProposal(dao, proposalId, data);
     }
 
+    /**
+     * @notice Processes a proposal that was sponsored.
+     * @dev Only members can process a proposal.
+     * @dev Only if the voting pass the proposal is processed.
+     * @dev Reverts when the adapter address is already in use and it is an adapter addition.
+     * @param dao The dao address.
+     * @param proposalId The guild kick proposal id.
+     */
     function processProposal(DaoRegistry dao, bytes32 proposalId)
         external
         override
+        onlyMember(dao)
```

Comment:
why should only members call that? What would happen if a non member processes a proposal?